File: 441-arranging-coins/441-arranging-coins.cpp
class Solution {
public:
    int arrangeCoins(int n) {
        int start(0), end(n);
        while(start < end)
        {
            long int mid = start + (end - start) / 2;
            long int k = mid * (mid + 1) / 2;
            if(k > n) end = mid;
            else start = mid + 1;
        }
        return n== 1 ? 1: end-1;
    }
};

File: 441-arranging-coins/441-arranging-coins.py
class Solution:
    def arrangeCoins(self, n: int) -> int:
        start,end = 0,n
        while start < end:
            mid = start + (end - start) // 2
            k = mid * (mid+1) // 2
            if k > n:
                end = mid
            else:
                start = mid + 1
        return 1 if n==1 else end - 1